vault-core/src/main/java/org/apache/jackrabbit/vault/fs/spi/impl/jcr20/JackrabbitACLManagement.java

```diff
@@ -202,18 +182,22 @@ private static boolean isRootNode(Node node) throws RepositoryException {
         return node.getDepth() == 0;
     }
 
-    private boolean areAuthorizablesAllowedBelowPath(Session session, String nodePath) throws RepositoryException {
-        if (usersRootPath == null || groupsRootPath == null) {
-            determineAuthorizableRootPaths(session);
-        }
-        return nodePath.startsWith(usersRootPath) || nodePath.startsWith(groupsRootPath);
+    private boolean isAuthorizable(Node node) throws RepositoryException {
+        return userManagement.isAuthorizableNodeType(node.getPrimaryNodeType().getName());
+    }
+
+    private boolean isAuthorizableFolder(Node node) throws RepositoryException {
+        return node.isNodeType(NT_REP_AUTHORIZABLE_FOLDER);
+    }
+
+    private boolean areAuthorizablesAllowedBelowPath(Node node) throws RepositoryException {
```

Comment:
```suggestion
    private boolean areAuthorizablesAllowed(Node node) throws RepositoryException {
```

as the argument is a node and not a path.